### DISCUSSION TITLE
Should we push images to the top 3 cloud providers?

### DISCUSSION THREAD

**Discussion by wholtz** [mamba-org/micromamba-docker#410] [Ideas]
Should we put our images into the registries of the top 3 cloud providers (AWS, GCP, Azure)? This would be contingent on being able to find ways to do this for free, such as being accepted into an open source support program.

Main motivation is that it could improve image fetching performance for people who primarily operate within one of those cloud platforms.

**Comment by maresb** [comment]
Sounds like a great idea! Are you aware of other projects that do this?

That said, most of these images are pretty lean, so while it would be nice-to-have, I'd be surprised if the current situation were problematic for anyone.

**Reply by wholtz** [reply]
Looking on AWS ERC there are other open source projects, but not a ton. AWS mirrors the Docker official images onto AWS ECR, so that pulls in all the big name open source images like openjdk and apache.

If you have a CI pipeline that is pulling in our image or using [Nextflow with Wave containers](https://www.nextflow.io/docs/latest/wave.html), then you might save a couple of second per invocation, but a single user may kick off 1000s of these jobs over time, so it could add up.

There is also the exposure from being findable on these container registries. Currently searching AWS ECR for "conda" brings up 17 results. None of these appear to be containers without pre-populated conda environments.

**Reply by wholtz** [reply]
And I just noticed that someone has copied a couple of our images directly to AWS - https://gallery.ecr.aws/genophore/mambaorg/micromamba 

But not all of them.
